### PR TITLE
Programming Language - Added Dart Complete Resources

### LINKED ISSUE #500: 💡 [FEATURE] - Add "Dart" Complete Resources
### Idea Contribution

- [X] I have read all the feature request issues.
- [X] I'm interested in working on this issue
- [X] I'm part of GSSOC organization

### Explain feature request

Dart, a client-optimized programming language for fast apps on any platform, has gained significant traction, especially with the rise of Flutter for mobile app development. Despite its growing popularity, our current documentation lacks comprehensive resources dedicated to Dart.

### Explain your solution

Dart Basics:
-Setting up the Dart environment: Installation and configuration for different platforms.
-Understanding Dart syntax and features: Variables, data types, functions, and control flow.
-Object-oriented programming in Dart: Classes, objects, interfaces, and inheritance.
-Functional programming in Dart: Lambdas, higher-order functions, and collections.

Advanced Dart:
-Asynchronous programming: Futures, async/await, and streams.
-Error handling: Exception handling and custom exceptions.
-Dart Standard Library: Exploring the standard library and its utilities.
-Interoperability: Using Dart with other languages and platforms.

Dart for Flutter Development:
-Setting up Dart for Flutter: Configuring a Flutter project with Dart.
-Flutter Widgets: Building UIs with Dart and Flutter widgets.
-State Management: Managing state in Flutter applications with Provider, Riverpod, etc.
-Navigation and Routing: Implementing navigation in Flutter apps.
-Dart and Flutter Plugins: Using and creating plugins for Flutter.

Dart for Web Development:
-Dart and HTML: Building web applications with Dart.
-Dart with AngularDart: Using Dart with Angular for web development.
-Server-side Dart: Building backend services with Dart and frameworks like Aqueduct or Shelf.

Performance and Optimization:
-Best Practices: Writing efficient Dart code.
-Profiling and Debugging: Tools and techniques for profiling and debugging Dart applications.

Deployment:
-Building and Deploying Dart Applications: Compiling Dart code and deploying applications.
-Continuous Integration and Delivery: Setting up CI/CD pipelines for Dart projects.

Dart Projects:
-Real-world applications: Examples of Dart in production projects.

Additional Resources:
Books, Courses, and Tutorials: Recommended books, online courses, and tutorials for learning Dart.
Community and Support: Useful links to Dart communities, forums, and support channels.

### Any alternative approaches/features

To enhance the learning experience, I suggest including essential resources such as recommended books, YouTube channels, and websites. Additionally, providing a beginner-friendly roadmap can help new developers navigate through the resources step-by-step.

### Additional Context

This feature request aims to enrich the Web Development section by providing extensive learning materials for Dart, similar to existing resources for other technologies. This will cater to the high demand for Dart knowledge in mobile development with Flutter, web development, and backend development, improving the overall utility of our documentation for developers at all levels.

### PR DESCRIPTION
**Pull request by Vaibhavsg17**
- I have added the All Resources for DART Programming Languages 

Closes #500 

### Checklist:
Before submitting your pull request, ensure that you have completed the following tasks: 

- [x] I have carefully reviewed and adhered to the [contributing guidelines](https://github.com/jfmartinz/ResourceHub/blob/main/CONTRIBUTING.md) before creating this pull request.
- [x] I followed the prescribed PR title template. _(Check this if you're adding a resource)_

> fill the bracket with x. Something like this [x]

### CODE REVIEW

File: Programming Languages/Dart/readme.md [hunk 1]
```diff
@@ -0,0 +1,525 @@
...
+
+#### Use Cases of Flutter:
+
+- **Mobile App Development:** Flutter is widely used for building cross-platform mobile apps for various industries including eCommerce, finance, healthcare, and entertainment.
+
+- **Web Development:** With the availability of Flutter for web, developers can now leverage their existing Flutter skills to build beautiful and responsive web applications.
+
+- **Desktop App Development:** Flutter's support for desktop platforms such as Windows, macOS, and Linux opens up opportunities for building desktop applications with a single codebase.
+
+- **Embedded Systems:** Flutter can also be used for developing embedded systems applications
```
Comment (jfmartinz) [inline Programming Languages/Dart/readme.md:386]:
```suggestion

```

**Review by jfmartinz — CHANGES_REQUESTED** [review]

**Comment by Vaibhavsg17** [comment]
@jfmartinz i have removed flutter part now check

**Comment by Vaibhavsg17** [comment]
@jfmartinz i have solved the issue now you can merge the pr

File: Programming Languages/Dart/readme.md [hunk 2]
```diff
@@ -0,0 +1,488 @@
...
+  - [Dart Collections](#dart-collections)
+  - [Object-Oriented Programming (OOP) in Dart](#object-oriented-programming-oop-in-dart)
+  - [Data Structures in Dart](#data-structures-in-dart)
+  - [Dart for Mobile App and Web Development](#dart-for-mobile-app-and-web-development)
+  - [Dart Packages and Libraries](#dart-packages-and-libraries)
+- [Tools and IDEs](#tools-and-ides)
+- [Books](#books)
+- [Documentation](#documentation)
+- [YouTube](#youtube)
+- [Community and Support](#community-and-support)
```
Comment (jfmartinz) [inline Programming Languages/Dart/readme.md:23]:
```suggestion
- [Community](#community)
```

File: Programming Languages/Dart/readme.md [hunk 3]
```diff
@@ -0,0 +1,488 @@
...
+        <td><a href="https://api.dart.dev/stable/2.14.4/dart-core/dart-core-library.html">dart:core</a></td>
+        <td>The core library of Dart, providing basic functionalities such as numbers, strings, collections, and exceptions.</td>
+    </tr>
+    <tr>
+        <td><a href="https://api.dart.dev/stable/2.14.4/dart-async/dart-async-library.html">dart:async</a></td>
+        <td>The async library of Dart, offering classes and functions for asynchronous programming, including Futures, Streams, and asynchronous error handling.</td>
+    </tr>
+</table>
+
+### Tools and IDEs
```
Comment (jfmartinz) [inline Programming Languages/Dart/readme.md:385]:
```suggestion
### Tools
```

File: Programming Languages/Dart/readme.md [hunk 4]
```diff
@@ -0,0 +1,488 @@
...
+
+<table>
+    <tr>
+        <th>Tool/IDE Name</th>
```
Comment (jfmartinz) [inline Programming Languages/Dart/readme.md:389]:
```suggestion
        <th>Resource Name</th>
```

File: Programming Languages/Dart/readme.md [hunk 5]
```diff
@@ -0,0 +1,488 @@
...
+        <td>An online code editor for experimenting with Dart and Flutter code snippets, providing an interactive learning experience without the need for setup.</td>
+    </tr>
+</table>
+
+
+### Books
+
+<table>
+    <tr>
+        <th>Book Title</th>
```
Comment (jfmartinz) [inline Programming Languages/Dart/readme.md:415]:
```suggestion
        <th>Resource Name</th>
```

File: Programming Languages/Dart/readme.md [hunk 6]
```diff
@@ -0,0 +1,488 @@
...
+        <td><a href="https://www.packtpub.com/product/dart-programming-by-example/9781785287627">Dart Programming by Example</a></td>
+        <td>A hands-on guide to learning Dart programming language through practical examples, covering key concepts and best practices.</td>
+    </tr>
+</table>
+
+### Documentation
+
+<table>
+    <tr>
+        <th>Resource</th>
```
Comment (jfmartinz) [inline Programming Languages/Dart/readme.md:436]:
```suggestion
        <th>Resource Name</th>
```

File: Programming Languages/Dart/readme.md [hunk 7]
```diff
@@ -0,0 +1,488 @@
...
+        <td>Official documentation for Dart's package manager, Pub, used for managing dependencies and sharing packages.</td>
+    </tr>
+</table>
+
+
+### YouTube
+
+<table>
+    <tr>
+        <th>Channel/Playlist</th>
```
Comment (jfmartinz) [inline Programming Languages/Dart/readme.md:458]:
```suggestion
        <th>Resource Name</th>
```

File: Programming Languages/Dart/readme.md [hunk 8]
```diff
@@ -0,0 +1,488 @@
...
+        <th>Description</th>
+    </tr>
+    <tr>
+        <td><a href="https://www.youtube.com/playlist?list=PLlxmoA0rQ-LyHW9voBdNo4gEEIh0Sj1wX">Dart Programming Tutorial for Beginners</a></td>
+        <td>A YouTube playlist offering a step-by-step tutorial series for beginners to learn Dart programming from scratch.</td>
+    </tr>
+</table>
+
+### Community and Support
```
Comment (jfmartinz) [inline Programming Languages/Dart/readme.md:467]:
```suggestion
### Community
```

File: Programming Languages/Dart/readme.md [hunk 9]
```diff
@@ -0,0 +1,488 @@
...
+
+<table>
+    <tr>
+        <th>Resource</th>
```
Comment (jfmartinz) [inline Programming Languages/Dart/readme.md:471]:
```suggestion
        <th>Resource Name</th>
```

**Review by jfmartinz — CHANGES_REQUESTED** [review]

**Review by jfmartinz — APPROVED** [review]

**Pull request merged**